polar-core/src/vm.rs

```diff
@@ -2853,13 +2929,9 @@ impl Runnable for PolarVirtualMachine {
             self.maybe_break(DebugEvent::Goal(goal.clone()))?;
         }
 
-        if self.log {
-            self.print("⇒ result");
-            if self.tracing {
-                for t in &self.trace {
-                    self.print(&format!("trace\n{}", t.draw(self)));
-                }
-            }
+        self.log_with(LogLevel::Trace, || &"⇒ result", &[]);
+        for t in &self.trace {
+            self.log_with(LogLevel::Trace, || format!("trace\n{}", t.draw(self)), &[]);
```

Comment:
This was previously gated by `self.tracing` — was removing the gate intentional?